core/src/main/java/org/mule/runtime/core/internal/policy/CompositeSourcePolicy.java

```diff
@@ -239,25 +194,13 @@ private MessagingException mapErrorInInnerExecution(MessagingException messaging
    * @return a {@link Publisher} that emits {@link SourcePolicySuccessResult} which contains the response parameters and the
    *         result event of the execution or a {@link SourcePolicyFailureResult} which contains the failure response parameters
    *         and the {@link MessagingException} thrown by the policy chain execution when processing completes.
-   * @throws Exception if there was an unexpected failure thrown by executing the chain.
    */
   @Override
   public Publisher<Either<SourcePolicyFailureResult, SourcePolicySuccessResult>> process(CoreEvent sourceEvent,
                                                                                          MessageSourceResponseParametersProcessor respParamProcessor) {
     return commonPolicy.process(sourceEvent, respParamProcessor);
   }
 
-  protected Supplier<Map<String, Object>> resolveErrorResponseParameters(MessagingException e) {
-    final Map<String, Object> originalFailureResponseParameters =
-        ((InternalEvent) e.getEvent()).getInternalParameter(POLICY_SOURCE_ORIGINAL_FAILURE_RESPONSE_PARAMETERS);
-
-    return () -> getParametersTransformer()
-        .map(parametersTransformer -> concatMaps(originalFailureResponseParameters,
-                                                 parametersTransformer
-                                                     .fromMessageToErrorResponseParameters(e.getEvent().getMessage())))
-        .orElse(originalFailureResponseParameters);
-  }
-
   private Map<String, Object> concatMaps(Map<String, Object> originalResponseParameters,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this method into "SourceWithPoliciesFluxObjectFactory". ([squid:S3398](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3398))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7763&issues=AWnLHoTkotIXnmsUntKD&open=AWnLHoTkotIXnmsUntKD)

[ISSUE_KEY]: AWnLHoTkotIXnmsUntKD